Fixes to broken code rendering

snekse opened the pull request:
There are a bunch of changes in here to fix various code rendering issues I ran into.  Almost all of the render fixing changes are in the last "WIP" commit [thought I was going to be able to get back to this sooner and not have that be my last commit :-) ]

This is a representative sample of posts that ran into rendering issues, each for a different reason.

- 2015-05-12-my-first-ratpack-app-what-i-learned
- 2013-06-05-redefining-the-service-layer-with-groovy-categories
- 2017-09-28-aggregate-services-into-a-single-swagger
- 2014-08-19-add-javascript-unit-tests-and-run-them-with-grails-test-app
- 2016-03-01-unit-testing-uri-based-grails-filters
- 2015-05-12-my-first-ratpack-app-what-i-learned
- 2015-04-21-interfacing-groovy-and-java-with-native-libraries
- 2013-05-23-accessing-grails-configuration-with-spring-property-placeholders

The content rendered with these code changes was copied over to the OPI website project in a branch.  

https://github.com/objectpartners/opi-website/pull/221

The previous content was maintained with an "old-" prefix in the route as the example below.

Old
https://deploy-preview-221--objectpartners.netlify.com/2017/12/21/old-use-a-terraform-wrapper-script-to-easily-manage-terraform-installations/
New
https://deploy-preview-221--objectpartners.netlify.com/2017/12/21/use-a-terraform-wrapper-script-to-easily-manage-terraform-installations/

Commit:
Fix 'it writes correct folder' test

  Prev version was expecting a `2017` directory to be created.
  Posts are stored together in a flat directory,
  with no nesting created off the post meta data.
  Meta data is only used in naming the file.
  URL routes will be built by Gatsby using frontmatter info
  to build a route from the slug and date metadata.

Commit:
Fix typo in "separator"

Commit:
Fix to-markdown test failing to extract 'markdown' prop from result

Commit:
Comment Turndown config to warn about Remark processing

  While debugging a test, I wasn't seeing the final markdown behave
  how I expected it to based on these settings.

  Doing some digging turned up Remark whacking some of the work
  that turndown does.  Made a note to make it easier to remember that.

Commit:
Make the Remark config match the Turndown config for Horizontal Rule

  Turndown was configued to use `___` but Remark was outputting `* * *`

Commit:
Make small improvements in the `to-markdown` tests.

  Some tests were not extracting the markdown property from the response
  that calling `toMarkdown` was giving them.

  Also trailing new line differences which we probably don't care about.

Commit:
WIP

snekse commented on src/util/to-markdown.ts:
```diff
@@ -28,12 +30,41 @@ export function isMarkdown(html: string): boolean {
+    markdown = markdown.replace('addjs="', 'addjs src="'); //Replace the one occurrence of missing ` src`
     const matches =
-      markdown.match(/(\\)?\[addjs\s*src="[^"\\]+(["\\\]]+)?/g) || [];
+      // TODO: This regex handles everything but the missing "src" problem above
```
:-)  Oops, this was the part I meant to clean up.

DSchau commented on src/index.ts:
```diff
@@ -23,6 +23,12 @@ export async function parse(input: string): Promise<any> {
   const posts = await parsePosts(base, authors);
   const presentations = await parsePresentations(base, authors);
 
+  const replacements =
```
Nit: kinda weird variable naming here (i.e. what is pv?) This is just counting the number of replacements for posts/presentations, right? Just for debugging purposes?

snekse replied:
Yeah, this is just in here for debugging purposes.  When I searched `addjs` in the `export.xml`, it wasn't even close to the number of matches we were doing in processing.  This was my way of measuring that.

pv, cv is just previous and current value.  If there's a cool es6 way to do `posts.sum(it.addjsReplacements, 0)` w/o reduce or lodash, let me know.  Always looking for way to write more readable code.

DSchau commented on src/util/__tests__/to-markdown.ts:
```diff
@@ -99,7 +101,54 @@ var a = 'b';
   });
+
+  test('it translates old style gist blocks', async () => {
+    const html =
```
Should probably be a fixture, but I 👍 this test!

snekse replied:
I'm not sure I know what a fixture is, but I agree :-)

DSchau replied:
Create `src/util/__fixtures__/markdown-example.ts`

```ts
export const markdownExample = `
<!-- the content goes here -->
`;
```

then in the test

```ts
import { markdownExample } from '../__fixtures__/markdown-example.ts';

// profit
```

snekse replied:
Yep, I Googled it.  Already working on it

DSchau commented on src/util/to-markdown.ts:
```diff
@@ -28,18 +30,49 @@ export function isMarkdown(html: string): boolean {
+
+    let i = 0;
+
     for (let match of matches) {
```
This may as well just be a traditional for loop if you're gonna keep track of an index :)

snekse replied:
I think I was planning on taking this code out once I had all of the matches accounted for.  I'm still missing like 5 for some reason.

DSchau commented on src/util/to-markdown.ts:
```diff
@@ -88,33 +122,37 @@ function getSrcFromAddJs(addjs: string) {
     .replace('github', 'githubusercontent')
     .replace(/\.js(\?file=)?/, '/raw/');
+  const replacementUrl = gistUrlMapper[origContentUrl];
+  return replacementUrl ? replacementUrl : origContentUrl;
```
`return replacementUrl || origContentUrl`;

DSchau reviewed (approved):
Looks good, left some comments!

Commit:
Addressing comments in the PR

snekse commented:
Addressed most of the comments w/ this commit 

https://github.com/DSchau/wordpress-to-markdown/pull/14/commits/fc763c2a35ee71cbb4fe78670b2e8fd8ad5ecd1d
